### PR TITLE
Using .ini file for config

### PR DESCRIPTION
**Pull request by richard-dds**
I apologize for the big PR. It's a little unfocused.

I added a new `config.ini` file in the root dir. When the app starts, this file is read and transformed into a dictionary that Flask can understand. It's totally possible that we'll end up with an entire config directory, but I figured I'd wait for more discussion before creating it.

I also refactored to use app factory pattern (http://flask.pocoo.org/docs/1.0/patterns/appfactories/),
which is the reason that so many files have been touched.

### CODE REVIEW

File: fundz/config.py [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
-DEBUG = bool(os.getenv('DEBUG', True))
-
-SQLALCHEMY_DATABASE_URI = os.getenv(
-    'DATABASE_URI',
-    'postgres://postgres:postgres@localhost/fundz'
-)
-SQLALCHEMY_TRACK_MODIFICATIONS = False
+def map_config(config):
+    return {
+        'ENV': config['default']['ENVIRONMENT'],
```
Comment (dandds) [inline fundz/config.py:3]: Do we need a default section, or would bare key/val pairs be okay? How would other sections work in this config model?

**Review by dandds — COMMENTED** [review]
Looks good! Just a question about the INI.

File: fundz/config.py (hunk 1 above)
Comment (dandds) [reply fundz/config.py:3]: Whelp, nevermind, `ConfigParser` won't parse it without headers. That's annoying.

Comment (richard-dds) [reply fundz/config.py:3]: Yeah, unfortunately those seem to be required.

**Review by dandds — APPROVED** [review]

**Pull request merged**